core/toolbox/category.js

```diff
@@ -0,0 +1,269 @@
+   * @type {!Blockly.ToolboxCategory.ClassConfig}
+   * @protected
+   */
+  this.classConfig_ = {
+    'container': 'blocklyToolboxCategory',
+    'row': 'blocklyTreeRow',
+    'icon': 'blocklyTreeIcon',
+    'label': 'blocklyTreeLabel',
+    'contents': 'blocklyToolboxContents',
+    'selected': 'blocklyTreeSelected',
+    'openIcon': 'blocklyTreeIconOpen',
+    'closedIcon': 'blocklyTreeIconClosed'
+  };
+
+  Blockly.utils.object.mixin(this.classConfig_, categoryDef['classConfig']);
+
+  /**
+   * Whether or not the category should display it's children.
+   * @type {boolean}
+   * @protected
+   */
+  this.expanded_ = categoryDef['expanded'] == 'true' || categoryDef['expanded'];
+
+  /**
+   * True if the category is visible, false otherwise.
+   * @type {boolean}
+   * @private
+   */
+  this.isVisible_ = true;
+
+  /**
+   * Parse the contents for this category.
+   * @type {string|
+   *        !Array<!Blockly.ToolboxItem>|
+   *        !Array<!Blockly.utils.toolbox.FlyoutItemDef>}
+   * @protected
+   */
+  this.contents_ = this.parseContents_(categoryDef, this.hasChildren_);
+};
+
+Blockly.utils.object.inherits(Blockly.ToolboxCategory,
+    Blockly.CollapsibleToolboxItem);
+
+/**
+ * @typedef {{
+ *            container:?string,
+ *            row:?string,
+ *            icon:?string,
+ *            label:?string,
+ *            contents:?string,
+ *            selected:?string,
+ *            openIcon:?string,
+ *            closedIcon:?string,
+ *          }}
+ */
+Blockly.ToolboxCategory.ClassConfig;
+
+/**
+ * The number of pixels to move the category over at each nested level.
+ * @type {number}
+ */
+Blockly.ToolboxCategory.NestedValue = 19;
+
+/**
+ * Parse the contents array depending on if the category has children, is a
+ * dynamic category, or if it's contents are meant to be shown in the flyout.
+ * @param {!Blockly.utils.toolbox.Category} categoryDef The information needed
+ *     to create a category.
+ * @param {boolean} hasChildren True if this category has subcategories, false
+ *     otherwise.
+ * @return {string|!Array<!Blockly.ToolboxItem>|
+ *          !Array<!Blockly.utils.toolbox.FlyoutItemDef>}
+ *     The contents for this category.
+ * @private
+ */
+Blockly.ToolboxCategory.prototype.parseContents_ = function(categoryDef,
+    hasChildren) {
+  var toolboxItems = [];
+  var contents = categoryDef['contents'];
+  if (hasChildren) {
+    for (var i = 0; i < contents.length; i++) {
+      var child = new Blockly.ToolboxCategory(contents[i], this.parentToolbox_, this);
+      toolboxItems.push(child);
+    }
+  } else if (categoryDef['custom']) {
+    toolboxItems = categoryDef['custom'];
+  } else {
+    toolboxItems = contents;
+  }
+
+  return toolboxItems;
+};
+
+/**
+ * Add either the colour or the style for a category.
+ * @param {!Blockly.utils.toolbox.Category} categoryDef The object holding
+ *    information on the category.
+ * @return {string} The hex color for the category.
```

Comment:
I think it'd be easier to stick to one spelling `colour` vs `color`.
Is this a pattern you're intentionally following to use `colour` except for in the phrase `hex color`?

Reply:
Good catch, I think these this is just a copy paste mistake.